traits/observers/_testing.py

```diff
@@ -47,9 +51,6 @@ def call_add_or_remove_notifiers(**kwargs):
     **kwargs
         New argument values to use instead.
     """
-    def dispatch_same(handler, event):
-        handler(event)
-
```

Comment:
For tests trying to remove notifiers, notifiers with different dispatcher are considered different. Instead of keep supplying a locally defined dispatcher to maintain equality, it is convenient to just use a constant one.